### PR TITLE
BP-28: Etcd as metadata store

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

This BP is exploring using Etcd as the metadata store.

### CODE REVIEW

**Comment by sijie** [comment]
@jvrao : https://github.com/sijie/bookkeeper/blob/19590b58a0aa54c9714b339ffedc2ded13ec4160/site/bps/BP-28-etcd-as-metadata-store.md

File: site/bps/BP-28-etcd-as-metadata-store.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  - *Watchers*: The zookeeper watcher is one time watcher, applications can’t reliably use it to get updates. In order to set a watcher, you have to read a znode or get children. Imagine such a use case, clients are watching a list of znodes (e.g. list of bookies), when those clients expire, they have to get the list of znodes in order to rewatch the list, even the list is never changed.
+  - The combination of session expires and watchers is often the root cause of critical zookeeper outages.
+
+This proposal is to explore other existing systems such as etcd as the metadata store. Using Etcd doesn't address concerns #1, however it might potentially
+address concern #2 and #3 to some extend. And if you are running bookkeeper in k8s, there is already an Etcd instance available. It can become easier to run
+bookkeeper on k8s if we can use Etcd as the metadata store.
```
Comment (jiazhai) [inline site/bps/BP-28-etcd-as-metadata-store.md]: nit: seems bring some line breaks because of copy paste?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by sijie** [comment]
Marked this BP as `accepted` with 1 +1s and 0 -1s.

**Comment by sijie** [comment]
The CI failure is unrelated. so ignore CI result to merge this BP content.

**Comment by sijie** [comment]
IGNORE CI

**Comment by codingwangqi** [comment]
I have implemented an Etcd metadata driver based on @sijie prototype #1114 . I will send out a PR for that.

**Pull request closed without merging**